### PR TITLE
Reflector prometheus

### PR DESCRIPTION
**Pull request by Pryz**
This change adds a Prometheus Handler to serve the metrics via an http handler. I've only added this handler on the reflector so far.

### CODE REVIEW

File: pkg/cmd/ctlstore/main.go [hunk 1]
```diff
@@ -275,10 +285,24 @@ func supervisor(ctx context.Context, args []string) {
...
+
+		var promHandler *prometheus.Handler
+		if len(reflectorConfig.MetricsBind) > 0 {
+			promHandler = &prometheus.Handler{}
+
+			http.Handle("/metrics", promHandler)
+
+			go func() {
+				events.Log("Serving Prometheus metrics on %s", reflectorConfig.MetricsBind)
+				http.ListenAndServe(reflectorConfig.MetricsBind, nil)
```
Comment (collinvandyck) [inline pkg/cmd/ctlstore/main.go]: Would recommend logging the error return here.

File: pkg/cmd/ctlstore/main.go [hunk 2]
```diff
@@ -275,10 +285,24 @@ func supervisor(ctx context.Context, args []string) {
 		if cliCfg.Shadow {
 			shadow = "true"
 		}
+
+		var promHandler *prometheus.Handler
```
Comment (collinvandyck) [inline pkg/cmd/ctlstore/main.go]: Seems like you're configuring prometheus for the supervisor here, not the reflector.  The supervisor does have a reflector but I imagine you want to add this to the `reflector` func.

Comment (Pryz) [reply pkg/cmd/ctlstore/main.go]: you're right :) fixed !

File: pkg/cmd/ctlstore/main.go (hunk 1 above)
Comment (Pryz) [reply pkg/cmd/ctlstore/main.go]: done

**Pull request merged**